python/bandpassUtils.py

```diff
@@ -0,0 +1,251 @@
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+from glob import glob
+from lsst.sims.photUtils import Bandpass
+
+# Input components:
+
+# $SYSENG_THROUGHPUTS_DIR / components
+#   camera /
+#      detector /   vendor[1,2] / ven[1,2]QE.dat, ven[1,2]Losses.dat  (multiply for a particular vendor, use min between vendors for 'average')
+#      lens[1,2,3] / l[1,2,3]Glass.dat, l[1,2,3]SBBAR.dat, l[1,2,3]S2BBAR.dat  (pass Glass through savitzky_golay and multiply by BBARs)
+#              l[1,2,3]Losses   / l[1,2,3]S(1,2)(Condensation,Contamination).dat   (multiply all)
+#      filters / [u,g,r,i,z,y]-bandResponse.dat
+#              filterLosses/    filterS(1,2)(Condensation,Contamination).dat (multiply all, for all filters)
+#   telescope /
+#       mirror[1,2,3] / m[1,2,3]Losses.dat, m[1,2,3]ProtAlIdeal.dat   (multiply ideal and loss for each mirror, multiply all mirrors)
+
+def setDefaultDirs(rootDir=None):
+    """
+    Returns a dictionary with the default directory locations of each component of the system throughput.
+    """
+    # Set SYSENG_THROUGHPUTS_DIR to the root dir for syseng_throughputs ('setup syseng_throughputs' will do this automatically)
+    defaultDirs = {}
+    if rootDir is None:
+        rootDir = os.getenv('SYSENG_THROUGHPUTS_DIR')
```

Comment:
Do you want to support people who have not use eups to setup the stack?  If not, should this be

rootDir = lsst.utils.getPackageDir('sysengh_throughputs')

?